packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -79,85 +84,105 @@ def get_metrics_prom(dcos_api_session, node):
                 assert metric_name in response.text
         check_zookeeper_metrics()
 
 
 def test_metrics_masters_cockroachdb(dcos_api_session):
     """Assert that CockroachDB metrics on masters are present."""
-    for master in dcos_api_session.masters:
+    if dcos_api_session.masters:
         expected_metrics = ['CockroachDB', 'ranges_underreplicated']
 
         @retrying.retry(wait_fixed=STD_INTERVAL, stop_max_delay=METRICS_WAITTIME)
         def check_cockroachdb_metrics():
-            response = get_metrics_prom(dcos_api_session, master)
+            response = get_metrics_prom(dcos_api_session, dcos_api_session.masters[0])
             for metric_name in expected_metrics:
                 assert metric_name in response.text
         check_cockroachdb_metrics()
 
 
 def test_metrics_masters_adminrouter(dcos_api_session):
     """Assert that Admin Router metrics on masters are present."""
-    for master in dcos_api_session.masters:
+    if dcos_api_session.masters:
         expected_metrics = [
             'dcos_component_name="Admin Router"',
             'nginx_vts',
         ]
 
         @retrying.retry(wait_fixed=STD_INTERVAL, stop_max_delay=METRICS_WAITTIME)
         def check_adminrouter_metrics():
-            response = get_metrics_prom(dcos_api_session, master)
+            response = get_metrics_prom(dcos_api_session, dcos_api_session.masters[0])
             for metric_name in expected_metrics:
                 assert metric_name in response.text
         check_adminrouter_metrics()
 
 
 def test_metrics_agents_adminrouter(dcos_api_session):
     """Assert that Admin Router metrics on agents are present."""
-    for master in dcos_api_session.slaves:
+    if dcos_api_session.slaves:
         expected_metrics = [
             'dcos_component_name="Admin Router Agent"',
             'nginx_vts',
         ]
 
         @retrying.retry(wait_fixed=STD_INTERVAL, stop_max_delay=METRICS_WAITTIME)
         def check_adminrouter_metrics():
-            response = get_metrics_prom(dcos_api_session, master)
+            response = get_metrics_prom(dcos_api_session, dcos_api_session.slaves[0])
             for metric_name in expected_metrics:
                 assert metric_name in response.text
         check_adminrouter_metrics()
 
+    if dcos_api_session.public_slaves:
+        expected_metrics = [
+            'dcos_component_name="Admin Router Agent"',
+            'nginx_vts',
+        ]
 
-def test_metrics_agents_statsd(dcos_api_session):
-    """Assert that statsd metrics on agent are present."""
-    if len(dcos_api_session.slaves) > 0:
+        @retrying.retry(wait_fixed=STD_INTERVAL, stop_max_delay=METRICS_WAITTIME)
+        def check_adminrouter_metrics():
+            response = get_metrics_prom(dcos_api_session, dcos_api_session.public_slaves[0])
+            for metric_name in expected_metrics:
+                assert metric_name in response.text
+        check_adminrouter_metrics()
```

Comment:
This `if` statement is copied from the above `if`; let's refactor so there isn't any code duplication.